internal/identity/identitymanager.go

```diff
@@ -0,0 +1,242 @@
+	}
+	im := &identityManager{
+		database:           di,
+		plugin:             ii,
+		blockchain:         bi,
+		identityCacheTTL:   config.GetDuration(config.IdentityManagerCacheTTL),
+		signingKeyCacheTTL: config.GetDuration(config.IdentityManagerCacheTTL),
+	}
+	// For the identity and signingkey caches, we just treat them all equally sized and the max items
+	im.identityCache = ccache.New(
+		ccache.Configure().MaxSize(config.GetInt64(config.IdentityManagerCacheLimit)),
+	)
+	im.signingKeyCache = ccache.New(
+		ccache.Configure().MaxSize(config.GetInt64(config.IdentityManagerCacheLimit)),
+	)
+
+	return im, nil
+}
+
+func (im *identityManager) GetLocalOrganization(ctx context.Context) (*fftypes.Organization, error) {
+	return im.cachedOrgLookupByAuthor(ctx, im.localOrgDID)
+}
+
+func (im *identityManager) OrgDID(org *fftypes.Organization) string {
+	return org.GetDID()
+}
+
+// ResolveInputIdentity takes in identity input information from an API call, or configuration load, and resolves
+// the combination
+func (im *identityManager) ResolveInputIdentity(ctx context.Context, identity *fftypes.Identity) (err error) {
+	log.L(ctx).Debugf("Resolving identity input: key='%s' author='%s'", identity.Key, identity.Author)
+
+	identity.Key, err = im.ResolveSigningKey(ctx, identity.Key)
+	if err != nil {
+		return err
+	}
+
+	// Resolve the identity
+	if err = im.resolveInputAuthor(ctx, identity); err != nil {
+		return err
+	}
+
+	log.L(ctx).Debugf("Resolved identity: key='%s' author='%s'", identity.Key, identity.Author)
+	return
+}
+
+func (im *identityManager) ResolveSigningKeyIdentity(ctx context.Context, signingKey string) (author string, err error) {
+
+	signingKey, err = im.ResolveSigningKey(ctx, signingKey)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: Consider other ways identity could be resolved
+	org, err := im.cachedOrgLookupBySigningKey(ctx, signingKey)
+	if err != nil {
+		return "", err
+	}
+
+	return im.OrgDID(org), nil
+
+}
+
+func (im *identityManager) ResolveLocalOrgDID(ctx context.Context) (localOrgDID string, err error) {
+	if im.localOrgDID != "" {
+		return im.localOrgDID, nil
+	}
+	orgKey := config.GetString(config.OrgKey)
+	if orgKey == "" {
+		orgKey = config.GetString(config.OrgIdentityDeprecated)
+	}
+	im.localOrgDID, err = im.ResolveSigningKeyIdentity(ctx, orgKey)
+	if err != nil {
+		return "", i18n.WrapError(ctx, err, i18n.MsgLocalOrgLookupFailed, orgKey)
+	}
+	if im.localOrgDID == "" {
+		return "", i18n.NewError(ctx, i18n.MsgLocalOrgLookupFailed, orgKey)
+	}
+	return im.localOrgDID, err
+}
+
+func (im *identityManager) ResolveSigningKey(ctx context.Context, inputKey string) (outputKey string, err error) {
+	// Resolve the signing key
+	if inputKey != "" {
+		if cached := im.signingKeyCache.Get(inputKey); cached != nil {
+			cached.Extend(im.identityCacheTTL)
+			outputKey = cached.Value().(string)
+		} else {
+			outputKey, err = im.blockchain.ResolveSigningKey(ctx, inputKey)
```

Comment:
I see that this resolution always goes to the blockchain plugin. In the context of tokens, I'm wondering how identities need to be expressed. For instance, as of #211, pool creation includes an operation to the token connector and an operation to the blockchain connector. It's not guaranteed that these connectors are talking to the same blockchain, so is it possible we'd need to resolve two keys via the two plugins? Maybe a question to ponder with @peterbroadhurst...

Reply:
I do think this is incompletely thought through for Tokens in this PR.
My suggestion is to address the token specifics in #215

Reply:
Yes, fine by me